qa/workunits/rbd/rbd_mirror_helpers.sh

```diff
@@ -1469,6 +1468,320 @@ wait_for_image_in_omap()
+        sleep ${s}
+        test_group_present \
+            "${cluster}" "${pool}" "${group}" "${state}" "${image_count}" &&
+        return 0
+    done
+    return 1
+}
+
+test_group_replay_state()
+{
+    local cluster=$1
+    local pool=$2
+    local group=$3
+    local test_state=$4
+    local image_count=$5
+    local status_result
+    local current_state=stopped
+
+    status_result=$(admin_daemons "${cluster}" rbd mirror group status ${pool}/${group} | grep -i 'state') || return 1
+    echo "${status_result}" | grep -i 'Replaying' && current_state=started
+    test "${test_state}" = "${current_state}" || return 1
+
+    test -n "${image_count}" || return 0
+
+    # TODO: test images when group image status is returned by the rbd-mirror
+}
+
+wait_for_group_replay_state()
+{
+    local cluster=$1
+    local pool=$2
+    local group=$3
+    local state=$4
+    local image_count=$5
+    local s
+
+    # TODO: add a way to force rbd-mirror to update replayers
+    for s in 0.1 1 2 4 8 8 8 8 8 8 8 8 16 16; do
+        sleep ${s}
+        test_group_replay_state "${cluster}" "${pool}" "${group}" "${state}" \
+                                "${image_count}" &&
+            return 0
+    done
+    return 1
+}
+
+wait_for_group_replay_started()
+{
+    local cluster=$1
+    local pool=$2
+    local group=$3
+    local image_count=$4
+
+    wait_for_group_replay_state "${cluster}" "${pool}" "${group}" started "${image_count}"
+}
+
+wait_for_group_replay_stopped()
+{
+    local cluster=$1
+    local pool=$2
+    local group=$3
+    local image_count=$4
+
+    wait_for_group_replay_state "${cluster}" "${pool}" "${group}" stopped "${image_count}"
+}
+
+get_newest_group_mirror_snapshot()
+{
+    local cluster=$1
+    local pool=$2
+    local image=$3
+    local log=$4
+
+    rbd --cluster "${cluster}" group snap list "${pool}/${group}" --format xml | \
+        xmlstarlet sel -t -c "//group_snaps/group_snap[state='ok' and position()=last()]" > \
+        ${log} || true
+}
+
+wait_for_group_snapshot_sync_complete()
+{
+    local local_cluster=$1
+    local cluster=$2
+    local pool=$3
+    local group=$4
+
+    local status_log=${TEMPDIR}/$(mkfname ${cluster}-${pool}-${group}.status)
+    local local_status_log=${TEMPDIR}/$(mkfname ${local_cluster}-${pool}-${group}.status)
+
+    mirror_group_snapshot "${cluster}" "${pool}" "${group}"
+    get_newest_group_mirror_snapshot "${cluster}" "${pool}" "${group}" "${status_log}"
+    local snapshot_name=$(xmlstarlet sel -t -v "//group_snap/snapshot" < ${status_log})
+
+    while true; do
+        for s in 0.2 0.4 0.8 1.6 2 2 4 4 8 8 16 16 32 32; do
+            sleep ${s}
+
+            get_newest_group_mirror_snapshot "${local_cluster}" "${pool}" "${image}" "${local_status_log}"
+            local local_snapshot_name=$(xmlstarlet sel -t -v "//group_snap/snapshot" < ${local_status_log})
+
+            test "${local_snapshot_name}" = "${snapshot_name}" && return 0
```

Comment:
Are the names of the snaps the same on both the clusters?  A better way would be to use the snapid for comparison as the snap names will probably need to be different on both clusters.

For instance, the corresponding function for images does the following:
 test "${snapshot_id}" = "${primary_snapshot_id}" && return 0